dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientBase.cs

```diff
@@ -316,22 +297,29 @@ private static ChatCompletionsOptions CreateChatCompletionsOptions(ChatRequestSe
             }
         }
 
-        foreach (ChatHistory.Message message in chat.Messages)
+        foreach (IChatMessage message in chatHistory)
         {
-            var role = message.AuthorRole switch
-            {
-                ChatHistory.AuthorRoles.User => ChatRole.User,
-                ChatHistory.AuthorRoles.Assistant => ChatRole.Assistant,
-                ChatHistory.AuthorRoles.System => ChatRole.System,
-                _ => throw new ArgumentException($"Invalid chat message author: {message.AuthorRole:G}")
-            };
+            ValidateChatAuthors(message.Role, out var validRole);
 
-            options.Messages.Add(new ChatMessage(role, message.Content));
+            options.Messages.Add(new ChatMessage(validRole, message.Content));
         }
 
         return options;
     }
 
+    private static void ValidateChatAuthors(string role, out ChatRole validRole)
+    {
+        Verify.NotNullOrWhiteSpace(role);
+
+        validRole = role.ToUpperInvariant() switch
```

Comment:
This is an odd way to do an `if` statement with multiple `string.Equals(...., StringComparison.OrdinalIgnoreCase)`